Source/WebGPU/WebGPU/ComputePipeline.h

```diff
@@ -73,18 +73,15 @@ class ComputePipeline : public WGPUComputePipelineImpl, public RefCounted<Comput
 
 private:
     ComputePipeline(id<MTLComputePipelineState>, const PipelineLayout&, MTLSize, Device&);
-    ComputePipeline(id<MTLComputePipelineState>, MTLComputePipelineReflection*, MTLSize, Device&);
+    ComputePipeline(id<MTLComputePipelineState>, Ref<PipelineLayout>, MTLSize, Device&);
     ComputePipeline(Device&);
 
     const id<MTLComputePipelineState> m_computePipelineState { nil };
 
-#if HAVE(METAL_BUFFER_BINDING_REFLECTION)
-    MTLComputePipelineReflection *m_reflection { nil };
-#endif
     const PipelineLayout *m_pipelineLayout { nullptr };
```

Comment:
Is there a possibility this is a UAF, if the pointed-to `PipelineLayout` is destroyed before the `ComputePipeline` is destroyed? Should both `m_pipelineLayout` and `m_generatedPipelineLayout` be unified into a single `Ref<>`?

Reply:
That is a good idea. I think we would need to drop the move semantics, since ComputePipeline would not be the sole owner of the pipeline layout. We could have multiple pipelines referencing the same pipeline layout, afaik.